File: algorithms/java/src/test/java/leetcode/LC217ContainsDuplicate.java
package leetcode;

import java.util.HashSet;
import java.util.Set;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * https://leetcode.com/problems/contains-duplicate/
 * #easy
 */
public class LC217ContainsDuplicate {
    public class Solution {
        public boolean containsDuplicate(int[] nums) {
            Set<Integer> seen = new HashSet<>();
            for (int num : nums) {
                if (seen.contains(num)) {
                    return true;
                }
                seen.add(num);
            }
            return false;
        }
    }

    @Test
    public void test_0_5_7() throws Exception {
        assertEquals(false, new Solution().containsDuplicate(new int[]{0, 5, 7}));
    }

    @Test
    public void test_0_5_7_5_10() throws Exception {
        assertEquals(true, new Solution().containsDuplicate(new int[]{0, 5, 7, 5, 10}));
    }
}

File: algorithms/code/leetcode/lc217_contains_duplicate/lc217_contains_duplicate.rs
// https://leetcode.com/problems/contains-duplicate/
// #easy

use std::collections::HashSet;

pub struct Solution;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut seen = HashSet::new();
        for num in nums {
            if seen.contains(&num) {
                return true;
            }
            seen.insert(num);
        }
        false
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_0_5_7() {
        assert!(!Solution::contains_duplicate(vec![0, 5, 7]));
    }

    #[test]
    fn test_0_5_7_5_10() {
        assert!(Solution::contains_duplicate(vec![0, 5, 7, 5, 10]));
    }
}
